sklearn/feature_extraction/_hashing.pyx

```diff
@@ -43,16 +42,24 @@ def transform(raw_X, Py_ssize_t n_features, dtype):
 
     for x in raw_X:
         for f, v in x:
-            value = v
-            if value == 0:
-                continue
-
             if isinstance(f, unicode):
                 f = f.encode("utf-8")
             # Need explicit type check because Murmurhash does not propagate
             # all exceptions. Add "except *" there?
             elif not isinstance(f, bytes):
                 raise TypeError("feature names must be strings")
+
+            if isinstance(v, unicode):
+                v = v.encode("utf-8")
+            if isinstance(v, (basestring, str, unicode)):
+                f = f + v
```

Comment:
We might as well make it directly equivalent to `DictVectorizer` by including the default '=' separator.